### PR TITLE
feat: Implement Feature Tag, Badge, Description and Owner APIs

### PR DESCRIPTION
**Pull request by allisonsuarez**
### Summary of Changes

- Implemented FeatureTagAPI, FeatureBadgeAPI, FeatureDescriptionAPI and FeatureOwnerAPIs
- Changed neo4j proxy behavior to reuse more general methods for owners

### Tests

- Added unit tests for all APIs
- Updated one test based on new behavior for neo4j
- Also manually tested all changes and new behavior

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: metadata/metadata_service/api/feature.py [hunk 1]
```diff
@@ -83,13 +86,25 @@ class FeatureOwnerAPI(Resource):
...
-    def delete(self, table_uri: str, owner: str) -> Iterable[Union[Mapping, int, None]]:
-        pass
+    @swag_from('swagger_doc/feature/owner_put.yml')
+    def put(self, feature_uri: str, owner: str) -> Iterable[Union[Mapping, int, None]]:
+        try:
+            self.client.add_resource_owner(uri=feature_uri, resource_type=ResourceType.Feature, owner=owner)
+            return {'message': f'The owner {owner} for feature_uri {feature_uri} '
+                               'was added successfully'}, HTTPStatus.OK
+        except Exception:
+            return {'message': f'The owner {owner} for feature_uri {feature_uri} was '
```
Comment (dkunitsk) [inline metadata/metadata_service/api/feature.py:96]: observation: bit odd there's no logging and err details are not returned. I know there also isn't for table owners.

File: metadata/metadata_service/proxy/atlas_proxy.py [hunk 2]
```diff
@@ -1456,3 +1456,26 @@ def get_lineage(self, *,
 
     def get_feature(self, *, feature_uri: str) -> Feature:
         pass
+
+    def get_resource_description(self, *,
```
Comment (dkunitsk) [inline metadata/metadata_service/proxy/atlas_proxy.py:1460]: I wonder what others think of the generality here. It raises the obvious future question of whether to do `get_resource_description` with Table as `resource_type` or use `get_table_description`.

Keeping with the per-entity boilerplate wagon seems more straightforward, perhaps.

File: metadata/metadata_service/api/feature.py (hunk 1 above)
Comment (sewardgw) [reply metadata/metadata_service/api/feature.py:96]: +1

File: metadata/metadata_service/proxy/atlas_proxy.py (hunk 2 above)
Comment (sewardgw) [reply metadata/metadata_service/proxy/atlas_proxy.py:1460]: One of the benefits we get from using the generic `get_resource_description` is that we can pretty much reuse all of the same code for any resource types with the one caveat being that they must all follow a similar data model within the DB. For instance, tables, dashboards, features, etc. must all have a "Description" and use the same relationship label. This way the # of one-off functions to support N number of resource types stays at 1 per action (e.g. get description, delete desc, etc.)

Comment (allisonsuarez) [reply metadata/metadata_service/proxy/atlas_proxy.py:1460]: I don't see why we would ever change this relationship for future entities so I think we should work towards making these methods general wherever we can.

File: metadata/metadata_service/api/feature.py (hunk 1 above)
Comment (allisonsuarez) [reply metadata/metadata_service/api/feature.py:96]: I'll add some!

**Review by dkunitsk — APPROVED** [review]
Great!

File: metadata/metadata_service/api/feature.py (hunk 1 above)
Comment (allisonsuarez) [reply metadata/metadata_service/api/feature.py:99]: done

File: metadata/metadata_service/proxy/atlas_proxy.py (hunk 2 above)
Comment (verdan) [reply metadata/metadata_service/proxy/atlas_proxy.py:1460]: +1 to have these functions generic enough to support all kinds of entities/resources, and the implementation can be different based on the `resource_type`

File: metadata/metadata_service/proxy/neo4j_proxy.py [hunk 3]
```diff
@@ -1452,7 +1485,7 @@ def get_dashboard_description(self, *,
         :return:
         """
 
-        return self._get_resource_description(resource_type=ResourceType.Dashboard, uri=id)
+        return self.get_resource_description(resource_type=ResourceType.Dashboard, uri=id)
 
     @timer_with_counter
     def put_dashboard_description(self, *,
```
Comment (verdan) [inline metadata/metadata_service/proxy/neo4j_proxy.py:1491]: How about moving calling `put_resource_description` directly from API with the `resource_type=ResourceType.Dashboard`?

**Review by verdan — APPROVED** [review]
just a few comments, the rest looks good to me.

File: metadata/metadata_service/proxy/neo4j_proxy.py (hunk 3 above)
Comment (allisonsuarez) [reply metadata/metadata_service/proxy/neo4j_proxy.py:1491]: This would be backwards incompatible so I decided to not do those changes in this PR

**Pull request merged**